readthedocs_ext/readthedocs.py

```diff
@@ -25,26 +24,25 @@
 
 try:
     # Available from Sphinx 2.0
-    from sphinx.builders.dirhtml import DirectoryHTMLBuilder
-    from sphinx.builders.html import StandaloneHTMLBuilder
     from sphinx.builders.singlehtml import SingleFileHTMLBuilder
 except ImportError:
-    from sphinx.builders.html import (DirectoryHTMLBuilder,
-                                      SingleFileHTMLBuilder,
-                                      StandaloneHTMLBuilder)
+    from sphinx.builders.html import SingleFileHTMLBuilder
 
 log = getLogger(__name__)
 
+
 DEFAULT_STATIC_URL = 'https://assets.readthedocs.org/static/'
+
+# Exclude the SingleHTML builder that is used by RTD to zip up local media
+# That builder is never used "online"
 ONLINE_BUILDERS = [
-    'readthedocs', 'readthedocsdirhtml', 'readthedocssinglehtml'
+    'html', 'dirhtml', 'singlehtml'
```

Comment:
I am worried this is going to break some of our users customizations, eg something like this: https://github.com/Invoca/developer-docs/blob/a1b05592d95fbdfee2f2360a3c2da9c427669c4c/source/conf.py#L158

I didn't see a ton of these in a GH search, and it's never been a properly supported way of finding out if you're running on RTD, but it's definitely going to break :/

Reply:
What's the alternative? If we want to use the regular Sphinx builders then people depending on us using a different builder are going to have something break on them.

Reply:
Another thing, sphinx creates tags with the name of the builder, users may be relying on that too (but probably this one is easy to fix). I'm on the phone, but I'll link to the issue and sphinxs docs in the morning.

Reply:
https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#directive-only

> The format and the name of the current builder (html, latex or text) are always set as a tag 4. To make the distinction between format and name explicit, they are also added with the prefix format_ and builder_, e.g. the epub builder defines the tags html, epub, format_html and builder_epub.
> These standard tags are set after the configuration file is read, so they are not available there.

And users already depending on that https://github.com/readthedocs/readthedocs.org/issues/4765#issuecomment-431927720

Reply:
If we want to perhaps we could deprecate the builders and roll using the stock ones out slowly. However I think we want to use the regular builders and not custom ones as much as possible.

The users in the linked issue should be ok since they're using ` .. only :: builder_html or readthedocs`

Reply:
Yea, I think there's no way around breaking this, but perhaps we should communicate it prior to doing it, but I don't know a great way to communicate it, since most users won't care and it's super deep and technical.

Reply:
I don't think there's a great way to communicate it either. It was never exactly supported so trying to tell people that the undocumented thing they were relying on is changing is hard. One possibility if we're really concerned about this is to keep the custom builders and have a feature flag that determines whether somebody uses the stock builders or the custom ones.

Reply:
Yea, I think we can probably just ship it and see what happens.